src/main/java/com/example/demo/ShortenUrlController.java

```diff
@@ -11,25 +13,27 @@
 public class ShortenUrlController {
 
     @Autowired
-    private UrlRepository repository;
+    private ShortenUrlService shortenUrlService;
 
     @GetMapping
-    public String helloWorld(){
-        return "Hello World from Spring Boot";
+    public ResponseEntity<String> helloWorld(){
+        String msg = "Hello World from Spring Boot";
+        HttpHeaders headers = new HttpHeaders();
+        return new ResponseEntity<String>(msg,headers, HttpStatus.OK);
     }
 
     @PostMapping("/url")
-    public Url saveUrl(@RequestBody Url url){
-
-        //TODO: Use ResponseEntity
-        return repository.save(url);
+    public ResponseEntity<Void> saveUrl(@RequestBody Url url){
+        shortenUrlService.saveUrl(url);
+        HttpHeaders headers = new HttpHeaders();
+        return ResponseEntity.status(HttpStatus.OK).headers(headers).build();
     }
 
     @GetMapping("/url")
-    public List<Url> getUrls(){
-
-        //TODO: Use ResponseEntity
-        return repository.findAll();
+    public ResponseEntity<List<Url>> getUrls(){
+        List<Url> urlList = shortenUrlService.getUrls();
```

Comment:
Could you please add all CRUD endpoint also

findUrlById
deleteUrlById(soft delete)
deleteAllUrls(soft delete)

And also I don't see expiration time of URL

Reply:
Should I use an attibute as "deleted" in the model to handle the soft delete? Or, is it a configuration in the mongo repository to do that?

Reply:
Discussed offline